### PR TITLE
Optimize generic-assay-meta endpoints with ClickHouse derived table

### PR DESCRIPTION
**Pull request by dippindots**
## Reason

The `generic-assay-meta` endpoints suffer from an N+1 query pattern and a 3-step SQL chain (profile → entity → meta + properties), resulting in O(N) database round-trips per request. For requests with many molecular profile IDs, this causes high latency and memory pressure.

### ClickHouse Schema
- Add `generic_assay_meta_derived` table with pre-aggregated [Map(String, String)](cci:2://file:///Users/zhaog/Desktop/Github/cbioportal/src/main/java/org/cbioportal/infrastructure/repository/clickhouse/generic_assay/ClickhouseGenericAssayMapper.java:17:0-89:1) properties column
- Add `generic_assay_profile_entity_derived` table for fast profile→entity resolution
- Add `ClickhouseMapTypeHandler` to map ClickHouse [Map(String, String)](cci:2://file:///Users/zhaog/Desktop/Github/cbioportal/src/main/java/org/cbioportal/infrastructure/repository/clickhouse/generic_assay/ClickhouseGenericAssayMapper.java:17:0-89:1) to Java `HashMap`
- Bump derived table version from 1.0.7 to 1.0.8

### Column-Store DDD Layer
- Add [GetGenericAssayMetaUseCase](cci:2://file:///Users/zhaog/Desktop/Github/cbioportal/src/main/java/org/cbioportal/domain/generic_assay/usecase/GetGenericAssayMetaUseCase.java:12:0-75:1) with `@Cacheable` and `TreeSet`-based cache key normalization
- Add [ColumnStoreGenericAssayController](cci:2://file:///Users/zhaog/Desktop/Github/cbioportal/src/main/java/org/cbioportal/application/rest/vcolumnstore/ColumnStoreGenericAssayController.java:34:0-125:1) at `/api/column-store/generic-assay-meta/*` (POST fetch, GET by profile, GET by stable ID)
- Add single [getGenericAssayMetaByProfileIds](cci:1://file:///Users/zhaog/Desktop/Github/cbioportal/src/main/java/org/cbioportal/infrastructure/repository/clickhouse/generic_assay/ClickhouseGenericAssayMapper.java:79:2-88:96) query that joins derived tables — reduces O(N) queries to 1 (0 on cache hit)
- Keep legacy [GenericAssayServiceImpl](cci:2://file:///Users/zhaog/Desktop/Github/cbioportal/src/main/java/org/cbioportal/legacy/service/impl/GenericAssayServiceImpl.java:23:0-273:1) and [GenericAssayController](cci:2://file:///Users/zhaog/Desktop/Github/cbioportal/src/main/java/org/cbioportal/legacy/web/GenericAssayController.java:31:0-138:1) unchanged (backward compatible)

### Streaming and Injection
- Switch [GenericAssayController](cci:2://file:///Users/zhaog/Desktop/Github/cbioportal/src/main/java/org/cbioportal/legacy/web/GenericAssayController.java:31:0-138:1) to `StreamingResponseBody` for lower memory usage
- Convert [GenericAssayController](cci:2://file:///Users/zhaog/Desktop/Github/cbioportal/src/main/java/org/cbioportal/legacy/web/GenericAssayController.java:31:0-138:1) to constructor injection

## Tests

- Add `ColumnStoreGenericAssayControllerTest` — controller-layer tests for all 3 endpoints
- Add `GetGenericAssayMetaUseCaseTest` — unit tests covering all branch paths (profiles-only, stableIds-only, both, null, ID projection)
- Add `ClickhouseMapTypeHandlerTest` — covers null, Map, non-String coercion, and unrecognized types
- Add `ClickhouseGenericAssayMapperTest` — 4 integration tests against testcontainer data
- Update `GenericAssayControllerTest` for `StreamingResponseBody` async dispatch
- Add 5 branch-coverage tests to [GenericAssayServiceImpTest](cci:2://file:///Users/zhaog/Desktop/Github/cbioportal/src/test/java/org/cbioportal/legacy/service/impl/GenericAssayServiceImpTest.java:28:0-330:1)

## Expected Results

- **Performance**: O(N) SQL queries reduced to 1 query (0 on cache hit) for multi-profile requests
- **Memory**: Streaming response avoids buffering full JSON body in memory
- **Backward compatibility**: Legacy `/api/generic_assay_meta/*` endpoints unchanged, still use MyBatis
- **Cache efficiency**: Normalized cache keys ensure same profile/stable IDs in different order share cache entries

### CODE REVIEW

File: src/main/java/org/cbioportal/legacy/service/impl/GenericAssayServiceImpl.java [hunk 1]
```diff
@@ -3,100 +3,92 @@
...
+  private final MolecularDataRepository molecularDataRepository;
 
-  @Autowired private SampleService sampleService;
+  private final SampleService sampleService;
 
-  @Autowired private MolecularProfileService molecularProfileService;
+  private final MolecularProfileService molecularProfileService;
 
-  @Autowired private SampleListRepository sampleListRepository;
+  private final SampleListRepository sampleListRepository;
```
Comment (alisman) [inline src/main/java/org/cbioportal/legacy/service/impl/GenericAssayServiceImpl.java]: @haynescd @onursumer @fuzhaoyuan hi guys, @dippindots says Sonar advised us to use constructor instead of autowire.  Since we are using autotwire elsewhere in the "clean architecture" i want to make sure we agree on what should be the convention for us.

File: src/test/java/org/cbioportal/domain/generic_assay/usecase/GetGenericAssayMetaUseCaseTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+  @Mock private GenericAssayRepository repository;
+
+  private List<GenericAssayMeta> createMockMetaList() {
+    GenericAssayMeta meta1 = new GenericAssayMeta(GENERIC_ASSAY_ID_1, ENTITY_TYPE);
+    HashMap<String, String> map1 = new HashMap<>();
+    map1.put(PROPERTY_NAME_1, PROPERTY_VALUE_1);
+    meta1.setGenericEntityMetaProperties(map1);
+
+    GenericAssayMeta meta2 = new GenericAssayMeta(GENERIC_ASSAY_ID_2, ENTITY_TYPE);
```
[the remark this answers is not in the document]
Comment (dippindots) [reply src/test/java/org/cbioportal/domain/generic_assay/usecase/GetGenericAssayMetaUseCaseTest.java]: good catch

File: src/main/java/org/cbioportal/domain/generic_assay/repository/GenericAssayRepository.java [hunk 3]
```diff
@@ -52,4 +53,35 @@ List<ClinicalDataCount> getGenericAssayDataBinCounts(
   List<GenericAssayDataCountItem> getGenericAssayDataCounts(
       StudyViewFilterContext studyViewFilterContext,
       List<GenericAssayDataFilter> genericAssayDataFilters);
+
+  /**
+   * Retrieves distinct generic assay entity stable IDs associated with the given molecular profile
+   * IDs via generic_assay_data_derived.
+   *
```
[the remark this answers is not in the document]
Comment (dippindots) [reply src/main/java/org/cbioportal/domain/generic_assay/repository/GenericAssayRepository.java:60]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/cBioPortal/cbioportal/pull/11979#discussion_r2865904395)

File: src/main/java/org/cbioportal/domain/generic_assay/usecase/GetGenericAssayMetaUseCase.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+   * @param projection projection level (e.g. "ID", "SUMMARY", "DETAILED")
+   * @return a list of {@link GenericAssayMeta}
+   */
+  @Cacheable(
+      cacheResolver = "generalRepositoryCacheResolver",
+      condition = "@cacheEnabledConfig.getEnabled()",
+      key =
+          "{#stableIds == null ? null : new java.util.TreeSet(#stableIds.?[#this != null]),"
+              + " #molecularProfileIds == null ? null : new java.util.TreeSet(#molecularProfileIds.?[#this != null]),"
+              + " #projection}")
```
Comment (onursumer) [inline src/main/java/org/cbioportal/domain/generic_assay/usecase/GetGenericAssayMetaUseCase.java:43]: this key function probably deserves a comment since it is not straightforward

File: src/main/java/org/cbioportal/application/rest/vcolumnstore/ColumnStoreGenericAssayController.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+ * <p>Provides endpoints to fetch generic assay meta data by molecular profile ID or stable ID,
+ * leveraging the {@link GetGenericAssayMetaUseCase} for the core logic.
+ */
+@Hidden
+@RestController
+@RequestMapping("/api/column-store")
+@Validated
+public class ColumnStoreGenericAssayController {
+
+  private final GetGenericAssayMetaUseCase getGenericAssayMetaUseCase;
```
Comment (onursumer) [inline src/main/java/org/cbioportal/application/rest/vcolumnstore/ColumnStoreGenericAssayController.java:41]: We can also define three distinct use cases (instead of calling the execute function with null values) but that may complicate the caching efforts.

File: src/main/java/org/cbioportal/infrastructure/repository/clickhouse/typehandlers/ClickhouseMapTypeHandler.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.ibatis.type.BaseTypeHandler;
+import org.apache.ibatis.type.JdbcType;
+
+/**
+ * MyBatis TypeHandler to convert ClickHouse Map(String, String) columns to Java {@code Map<String,
+ * String>}.
+ */
+public class ClickhouseMapTypeHandler extends BaseTypeHandler<Map<String, String>> {
```
Comment (onursumer) [inline src/main/java/org/cbioportal/infrastructure/repository/clickhouse/typehandlers/ClickhouseMapTypeHandler.java:16]: It's surprising to me that we have to manually implement a mapper to map a map to a map :)

File: src/main/java/org/cbioportal/infrastructure/repository/clickhouse/typehandlers/ClickhouseMapTypeHandler.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Override
+  public Map<String, String> getNullableResult(CallableStatement cs, int columnIndex)
+      throws SQLException {
+    return convertToMap(cs.getObject(columnIndex));
+  }
+
+  private Map<String, String> convertToMap(Object value) {
+    if (value instanceof Map<?, ?> map) {
+      HashMap<String, String> result = new HashMap<>();
```
Comment (onursumer) [inline src/main/java/org/cbioportal/infrastructure/repository/clickhouse/typehandlers/ClickhouseMapTypeHandler.java]: Not sure if we need it here but using `LinkedHashMap` would make sure a deterministic order of data values

File: src/main/java/org/cbioportal/infrastructure/repository/clickhouse/typehandlers/ClickhouseMapTypeHandler.java (hunk 6 above)
Comment (dippindots) [reply src/main/java/org/cbioportal/infrastructure/repository/clickhouse/typehandlers/ClickhouseMapTypeHandler.java:16]: agree, I had to make this customize converter in mybatis to make it work...

File: src/main/java/org/cbioportal/infrastructure/repository/clickhouse/typehandlers/ClickhouseMapTypeHandler.java (hunk 7 above)
Comment (dippindots) [reply src/main/java/org/cbioportal/infrastructure/repository/clickhouse/typehandlers/ClickhouseMapTypeHandler.java]: will change to `LinkedHashMap`

File: src/main/java/org/cbioportal/domain/generic_assay/usecase/GetGenericAssayMetaUseCase.java (hunk 4 above)
Comment (dippindots) [reply src/main/java/org/cbioportal/domain/generic_assay/usecase/GetGenericAssayMetaUseCase.java:43]: added comments for this logic
